core/trino-spi/src/main/java/io/trino/spi/function/Signature.java

```diff
@@ -280,24 +272,7 @@ public Builder variableArity()
 
         public Signature build()
         {
-            return fromJson(name, typeVariableConstraints, longVariableConstraints, returnType, argumentTypes, variableArity);
+            return new Signature(name, typeVariableConstraints, longVariableConstraints, returnType, argumentTypes, variableArity);
         }
     }
-
-    /**
-     * This method is only visible for JSON deserialization.
-     * @deprecated use builder
-     */
-    @Deprecated
-    @JsonCreator
-    public static Signature fromJson(
```

Comment:
cc @dain

Reply:
This method is here on purpose to make it clear that this method is only here for Jasckson json parsing because we disable access to private constructors when we use JSON.  If the parsing works when this is removed, it implies that the code deserializing the JSON has a misconfigured parser.